spec/models/product_spec.rb

```diff
@@ -1,509 +1,557 @@
-    context 'when a user is a product user' do
-      it 'finds the product_user' do
-        expect(product.find_product_user(user)).to eq product_user
+        it "is invalid" do
+          expect(new_product).to_not be_valid
+          expect(new_product.errors.messages).to include :url_name
+        end
       end
-    end
 
-    context 'when a user is not a product user' do
-      let(:other_user) { create(:user) }
-
-      it 'does not find a product_user' do
-        expect(product.find_product_user(other_user)).to be_nil
-      end
-    end
-  end
+      context "when a product's url_name exists in another facility" do
+        let(:other_facility) { create(:facility) }
+        let(:new_product) { build(:instrument_requiring_approval, url_name: "product_name", facility: facility) }
 
-  context '#has_access_list?' do
-    context 'when its type supports access groups' do
-      context 'when it has an access group' do
         before :each do
-          product.product_access_groups = [ access_group ]
+          other_facility.facility_accounts.create(attributes_for(:facility_account))
+          create(:instrument_requiring_approval, url_name: "product_name", facility: other_facility)
         end
 
-        it 'has an access list' do
-          expect(product.has_access_list?).to be true
+        it "is valid" do
+          expect(new_product).to be_valid
         end
       end
+    end
+  end
 
-      context 'when it has no access groups' do
-        it 'does not have an access list' do
-          expect(product.has_access_list?).to be false
-        end
-      end
+  describe "#training_request_contacts" do
+    let(:product) { build(:item, training_request_contacts: contacts) }
+    subject(:emails) { product.training_request_contacts.to_a }
+
+    describe "blank" do
+      let(:contacts) { "" }
+      it { is_expected.to eq [] }
     end
 
-    context 'when its type does not support access groups' do
-      let(:generic_item) { build(:setup_item) }
+    describe "nil" do
+      let(:contacts) { nil }
+      it { is_expected.to eq [] }
+    end
 
-      it 'does not have an access list' do
-        expect(generic_item.has_access_list?).to be false
-      end
+    describe "a single address" do
+      let(:contacts) { "testing@example.com" }
+      it { is_expected.to eq ["testing@example.com"] }
     end
-  end
 
-  context "url_name collisions" do
-    context "when a product's url_name exists in its facility" do
-      let(:new_product) { build(:instrument_requiring_approval, url_name: product.url_name, facility: product.facility) }
+    describe "multiple addresses" do
+      let(:contacts) { "testing@example.com, test2@nucore.com" }
+      it { is_expected.to eq ["testing@example.com", "test2@nucore.com"] }
+    end
 
-      it "is invalid" do
-        expect(new_product).to_not be_valid
-        expect(new_product.errors.messages).to include :url_name
-      end
+    describe "some extra commas in there" do
```

Comment:
Made an update that validates each of the addresses. I went with a pretty simple regex. We have a more complicated one inside `User`, but I didn't want to go crazy with it.